### PR TITLE
2382 support yagop contribution of rpc endpoint get mempool pending operations

### LINKED ISSUE #2382: Fetch transactions pending in Mempool
**Is your feature request related to a problem and use case? Please describe.**
RPC method to fetch transactions from mempool is missing

**Describe the solution you'd like**
Implement method to fetch transactions from mempool

**Describe alternatives you've considered**
Using other library or implementing my own

**Additional context**
Some transactions fail due to the address being used. I need to query the mempool to avoid broadcasting tx in the same block

### PR DESCRIPTION
**Pull request by hui-an-yang**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [x] Your code builds cleanly without any errors or warnings
- [x] You have run the linter against the changes
- [x] You have added unit tests (if relevant/appropriate)
- [x] You have added integration tests (if relevant/appropriate)
- [x] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [x] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [x] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [x] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

support rpc endpoint to get pending_operations from mempool
closes #2382

### CODE REVIEW

File: packages/taquito-rpc/test/data/rpc-responses.ts [hunk 1]
```diff
@@ -5404,6 +5404,109 @@ export const ticketBalancesResponse = [
   },
 ];
 
+export const pendingOperationsResponse = {
```
Comment (yagop) [inline packages/taquito-rpc/test/data/rpc-responses.ts:5407]: Would be nice if the team add type check on `createRequest` as it merely does a conversion from `any` to any type of the template is provided.

As `createRequest` lacks type check I typed this constant as `MempoolPendingOperationsResponse` to enforce the test to do it.

May you add the `MempoolPendingOperationsResponse` type back?

Comment (hui-an-yang) [reply packages/taquito-rpc/test/data/rpc-responses.ts:5407]: I renamed type `MempoolPendingOperationsResponse` to `PendingOperations` in packages/taquito-rpc/src/types.ts line 2138 to separate it from the mock rpc response aka pendingOperationsResponse here. And have pushed a commit to use the type on the rpc call in tests to enforce it.

File: packages/taquito-contracts-library/src/rpc-wrapper.ts [hunk 2]
```diff
@@ -46,6 +46,8 @@ import {
   VotingPeriodBlockResult,
   TicketTokenParams,
   AllTicketBalances,
+  PendingOperations,
+  PendingOperationsQuery,
```
Comment (zainen) [inline packages/taquito-contracts-library/src/rpc-wrapper.ts:50]: should this follow the same conventions of `params`?

File: packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts [hunk 3]
```diff
@@ -1224,4 +1227,25 @@ export class RpcClientCache implements RpcClientInterface {
       return response;
     }
   }
+
+  /**
+   * @description List the prevalidated operations in mempool (accessbililty of mempool depends on each rpc endpoint)
+   * @param args has 5 optional properties. We support version 1 with new encoding as version 0 will be deprecated soon. The rest of the properties is to filter pending operatoins response
+   * @default args { version: '1', applied: true, refused: true, outdated, true, branchRefused: true, branchDelayed: true, validationPass: undefined }
```
Comment (zainen) [inline packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts:1234]: typo `pending operatoins response`

File: packages/taquito-rpc/src/taquito-rpc.ts [hunk 4]
```diff
@@ -1172,4 +1174,18 @@ export class RpcClient implements RpcClientInterface {
       method: 'GET',
     });
   }
+
+  /**
+   * @description List the prevalidated operations in mempool (accessbililty of mempool depends on each rpc endpoint)
+   * @param args has 5 optional properties. We support version 1 with new encoding as version 0 will be deprecated soon. The rest of the properties is to filter pending operatoins response
```
Comment (zainen) [inline packages/taquito-rpc/src/taquito-rpc.ts:1180]: same typo

**Review by zainen — CHANGES_REQUESTED** [review]

File: packages/taquito-contracts-library/src/rpc-wrapper.ts (hunk 2 above)
Comment (hui-an-yang) [reply packages/taquito-contracts-library/src/rpc-wrapper.ts:50]: it's a url query that looks like `?key=value` tailing a url. There so the name but will update it to follow the convention of `BakingRightsQueryArguments` and `EndorsingRightsQueryArguments`

File: packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts (hunk 3 above)
Comment (zainen) [reply packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts:1234]: this one is still here

Comment (hui-an-yang) [reply packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts:1234]: Are we seeing differently?
image

File: packages/taquito-rpc/test/data/rpc-responses.ts (hunk 1 above)
Comment (yagop) [reply packages/taquito-rpc/test/data/rpc-responses.ts:5407]: Perfect! 👍

**Review by yagop — APPROVED** [review]

File: packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts (hunk 3 above)
Comment (zainen) [reply packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts:1234]: I dont see it as outdated and if i go to the file in the PR its still there :O 
i have had this a couple times too recently...

Comment (zainen) [reply packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts:1234]: coupld be because of the rebase problem

Comment (zainen) [reply packages/taquito-rpc/src/rpc-client-modules/rpc-cache.ts:1234]: nevermind 👀 i just cant follow the link attached to this comment

File: packages/taquito-rpc/src/taquito-rpc.ts [hunk 5]
```diff
@@ -1172,4 +1174,20 @@ export class RpcClient implements RpcClientInterface {
       method: 'GET',
     });
   }
+
+  /**
+   * @description List the prevalidated operations in mempool (accessbililty of mempool depends on each rpc endpoint)
```
Comment (dsawali) [inline packages/taquito-rpc/src/taquito-rpc.ts:1179]: typo

File: packages/taquito-rpc/src/types.ts [hunk 6]
```diff
@@ -2117,3 +2117,29 @@ export interface TxRollupInboxResponse {
...
+  version?: '1';
+  applied?: boolean;
+  refused?: boolean;
+  outdated?: boolean;
+  branchRefused?: boolean;
+  branchDelayed?: boolean;
+  validationPass?: '0' | '1' | '2' | '3';
+}
+
+type failedProcessedOperation = Pick<
```
Comment (dsawali) [inline packages/taquito-rpc/src/types.ts:2131]: type shouldn't have lowercase beginning, it should be UpperCamelCase

File: packages/taquito-rpc/test/data/rpc-responses.ts [hunk 7]
```diff
@@ -5404,6 +5404,109 @@ export const ticketBalancesResponse = [
...
+          kind: 'Preendorsement',
+          id: 'proto.016-PtMumbai.validate.consensus_operation_for_old_level',
+          expected: 128096,
+          provided: 128095,
+        },
+      ],
+    },
+  ],
+  branch_refused: [],
+  branch_dealyed: [],
```
Comment (dsawali) [inline packages/taquito-rpc/test/data/rpc-responses.ts:5506]: typo

**Review by dsawali — CHANGES_REQUESTED** [review]

**Review by dsawali — APPROVED** [review]

**Review by zainen — APPROVED** [review]
😄

**Pull request merged**